python/ray/autoscaler/_private/commands.py

```diff
@@ -416,9 +424,13 @@ def run_docker_stop(node, container_name):
 
     container_name = config.get("docker", {}).get("container_name")
     if container_name:
+        threads = []
         for node in A:
-            run_docker_stop(node, container_name)
-
+            t = Thread(target=lambda: run_docker_stop(node, container_name))
+            t.start()
```

Comment:
can you remove the `t.start()` to a for loop outside?
e.g.
```
for t in threads:
    t.start()
for t in threads:
    t.join()
```

Reply:
OOC, why is this preferred?

Reply:
This seems to be a convention for using it (you create the threads, then start them, then join them).
BTW, I just tested threading locally, and it seems that with 10000 threads this crashes. Should we do something like:
batch size = 1000
for batch in num_workers/batch_size -> for  worker in batch size->create thread?